MoreLinq/CountDown.cs

```diff
@@ -84,7 +84,7 @@ IEnumerable<TResult> IterateCollection(int i)
 
             IEnumerable<TResult> IterateSequence()
             {
-                var queue = new Queue<T>();
+                var queue = new Queue<T>(count + 1);
```

Comment:
This is problematic when count is negative (more specifically, less than -1) because `CountDown` permits a negative count. In fact, this highlighted that we're missing a test and that I've added in bb05a54e4fd1d00aef48ac830481e312702555be. It now demonstrates that this change [breaks](https://ci.appveyor.com/project/raboof/morelinq/build/1076#L596):

```
1) Error : MoreLinq.Test.CountDownTest.WithNegativeCount
System.ArgumentOutOfRangeException : Non-negative number required.
Parameter name: capacity
Actual value was -999.
   at System.Collections.Generic.Queue`1..ctor(Int32 capacity)
   at MoreLinq.MoreEnumerable.<>c__DisplayClass28_0`2.<<CountDown>g__IterateSequence|2>d.MoveNext()
   at NUnit.Framework.Constraints.NUnitEqualityComparer.EnumerablesEqual(IEnumerable x, IEnumerable y, Tolerance& tolerance)
   at NUnit.Framework.Constraints.NUnitEqualityComparer.AreEqual(Object x, Object y, Tolerance& tolerance)
   at NUnit.Framework.Constraints.EqualConstraint.ApplyTo[TActual](TActual actual)
   at NUnit.Framework.Assert.That[TActual](TActual actual, IResolveConstraint expression, String message, Object[] args)
   at NUnit.Framework.Assert.That[TActual](TActual actual, IResolveConstraint expression)
   at MoreLinq.Test.TestExtensions.AssertSequenceEqual[T](IEnumerable`1 actual, IEnumerable`1 expected)
   at MoreLinq.Test.CountDownTest.WithNegativeCount()
```